classes/01class/exercises/c01-aws02/jayanath/README.md

```diff
@@ -0,0 +1,94 @@
+# C01-AWS02
+
+## Commands Execution Output
+
+- Commands to create the S3 Bucket and to copy the file to the bucket:
+```
+// creating the bucket with LocationConstraint as our region is not us-east-1
+> aws s3api create-bucket --bucket devops-academy-rocks-9930.com \
+    --region ap-southeast-2 \
+    --create-bucket-configuration LocationConstraint=ap-southeast-2
+
+{
+    "Location": "http://devops-academy-rocks-9930.com.s3.amazonaws.com/"
+}
+
+// don't want to have a public bucket :-)
+> aws s3api put-public-access-block \
+    --bucket devops-academy-rocks-9930.com \
+    --public-access-block-configuration "BlockPublicAcls=true,IgnorePublicAcls=true,BlockPublicPolicy=true,RestrictPublicBuckets=true"
+
+// uploading some coffee to the bucket
+> aws s3 cp coffee.json s3://devops-academy-rocks-9930.com
+upload: ./coffee.json to s3://devops-academy-rocks-9930.com/coffee.json
+
+```
+
+- Commands to allow the EC2 instance to access the files in S3:
+```
+// need a bucket policy to provide access to our account
+// using CanonicalUser ID to avoid account ID on the repo in plain text
+> aws s3api put-bucket-policy --bucket devops-academy-rocks-9930.com --policy file://devops-bucket-policy.json
+    devops-bucket-policy.json content:
+    {
+    "Statement": [
+        {
+            "Effect": "Allow",
+            "Principal": {
+                "CanonicalUser": "2ea65f81fa3eb67b5cc6ca452212d4496fe3c05346d024356a755db3b2738895"
+            },
+            "Action": [
+                "s3:GetObject"
+            ],
+            "Resource": "arn:aws:s3:::devops-academy-rocks-9930.com/*"
+        }
+    ]
+    }
+
```

Comment:
if your instance have a instance profile attached to it that contains the required permission to execute the actions on S3, there is no need to also have a bucket policy in place as well.
You can have both, but for this case, it would not be necessary.
You normally use both when you have more complex rules, specially with permissions based on object and not to the entire bucket.

Reply:
Thanks Kiko. I see what you mean.